src/main/java/org/gridsuite/gateway/GatewayConfig.java

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.gridsuite.gateway;
+
+import com.auth0.jwt.interfaces.DecodedJWT;
+import com.nimbusds.openid.connect.sdk.claims.IDTokenClaimsSet;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.gateway.filter.GatewayFilterChain;
+import org.springframework.cloud.gateway.filter.GlobalFilter;
+import org.springframework.cloud.gateway.route.RouteLocator;
+import org.springframework.cloud.gateway.route.builder.RouteLocatorBuilder;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.stereotype.Component;
+import org.springframework.web.server.ServerWebExchange;
+import reactor.core.publisher.Mono;
+
+import java.util.List;
+
+import java.net.*;
+import com.nimbusds.jose.*;
+import com.nimbusds.jwt.*;
+import com.nimbusds.oauth2.sdk.id.*;
+import com.nimbusds.openid.connect.sdk.validators.*;
+
+/**
+ * @author Chamseddine Benhamed <chamseddine.benhamed at rte-france.com>
+ */
+@Configuration
+@EnableConfigurationProperties(UriConfiguration.class)
+public class GatewayConfig {
+    private static final Logger LOGGER = LoggerFactory.getLogger(GatewayConfig.class);
+
+    private final GatewayService gatewayService;
+    private boolean ignoreTokenValidation;
+
+    @Value("${allowed-issuers}")
+    private List<String> allowedIssuers;
+
+    public GatewayConfig(GatewayService gatewayService, UriConfiguration uriConfiguration) {
+        this.gatewayService = gatewayService;
+        this.ignoreTokenValidation = uriConfiguration.getIgnoreTokenValidation();
+    }
+
+    @Component
+    public class TokenValidatorGlobalPreFilter implements GlobalFilter {
+        @Override
+        public Mono<Void> filter(ServerWebExchange exchange, GatewayFilterChain chain) {
+            if (ignoreTokenValidation) {
```

Comment:
I think we said we would remove this ?

Reply:
no that's just for the test

Reply:
It's a bad design to have features used only to test inside non test code. We must refactor to avoid this I think